src/Npgsql/NpgsqlCommand.cs

```diff
@@ -125,7 +125,7 @@ public override string CommandText
             set
             {
                 _commandText = State == CommandState.Idle
-                    ? value
+                    ? value ?? string.Empty
```

Comment:
Oh... What the hell is going in ADO.NET?

By the way, is it possible that `_commandText` could be `null`? If so, makes sense to change the getter.

Reply:
I don't think _commandText could (or should) be null - we should initialize it and normalize it any attempt to set it to null into an empty string. Right now the nullability annotations don't reflect this because I didn't have a clear picture at the time of how it should work.

Reply:
SQL Client uses coalescing in the getter. Should we follow this way too?

Reply:
I don't think so... I'm a big believer of normalization/sanitization/coalescing as early as possible... Better to get rid of the null in the setter and not think about it again.